### PR TITLE
More RPS work

### PR DESCRIPTION
**Pull request by tomclose**

### CODE REVIEW

File: packages/rps/src/redux/game-v2/__tests__/player-a.test.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+describe('when joining an open game', () => {
+  it('calls createChannel and transitions to GameChosen', async () => {
+    const initialState = gameState(localStatesA.lobby);
+    const action = joinOpenGame(bName, bAddress, stake);
+    const state = channelStates.preFund0;
+
+    const { storeState } = await expectSaga(gameSaga as any, client)
+      .withReducer(reducer, initialState)
+      .dispatch(action)
+      .provide([callCreateChannel(state)])
```
Comment (geoknee) [inline packages/rps/src/redux/game-v2/__tests__/player-a.test.ts]: It looks like `callCreateChannel(state)` returns an array. Do you need to put that array in another array here?

File: packages/rps/src/redux/game-v2/__tests__/scenarios.ts [hunk 2]
```diff
@@ -78,7 +80,7 @@ const propsA = {
   theirWeapon: playerBWeapon,
 };
 
-export const localStatesA: Record<string, LocalState> = {
```
Comment (geoknee) [inline packages/rps/src/redux/game-v2/__tests__/scenarios.ts]: Why did you remove this type?

File: packages/rps/src/redux/game-v2/saga.ts [hunk 3]
```diff
@@ -1,205 +1,267 @@
...
-          const state = yield call(rpsChannelClient.closeChannel, channelId);
-          yield put(updateChannelState(state));
+    const { localState, channelState }: ls.GameState = yield select(getGameState);
+
+    if (cs.isClosed(channelState) && localState.type !== 'GameOver') {
+      yield* transitionToGameOver(localState, channelState);
+      continue;
+    }
+
+    switch (localState.type) {
```
Comment (geoknee) [inline packages/rps/src/redux/game-v2/saga.ts]: 👍

File: packages/rps/src/redux/game-v2/__tests__/player-a.test.ts (hunk 1 above)
Comment (tomclose) [reply packages/rps/src/redux/game-v2/__tests__/player-a.test.ts]: Yeah. You can `provide` many stubs at once, and each one is an array, for some reason.

File: packages/rps/src/redux/game-v2/__tests__/scenarios.ts (hunk 2 above)
Comment (tomclose) [reply packages/rps/src/redux/game-v2/__tests__/scenarios.ts]: Because I discovered it was actually removing type-safety: I could do `localStatesA.blah` and it wouldn't complain. Also, now I'm using constructors, typescript knows that all the objects are `LocalState`s anyway.
